src/RemoteRequest/CachedResponse.php

```diff
@@ -0,0 +1,117 @@
+<?php
+/**
+ * Class CachedResponse.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP\RemoteRequest;
+
+use DateTimeImmutable;
+use DateTimeInterface;
+
+/**
+ * Serializable object that represents a cached response together with its expiry time.
+ *
+ * @package AmpProject\AmpWP
+ */
+final class CachedResponse {
+
+	/**
+	 * Cached body.
+	 *
+	 * @var string
+	 */
+	private $body;
+
+	/**
+	 * Cached headers.
+	 *
+	 * @var array
+	 */
+	private $headers;
+
+	/**
+	 * Cached status code.
+	 *
+	 * @var int
+	 */
+	private $status_code;
+
+	/**
+	 * Expiry time of the cached value.
+	 *
+	 * @var DateTimeInterface
+	 */
+	private $expiry;
+
+	/**
+	 * Instantiate a CachedResponse object.
+	 *
+	 * @param string            $body         Cached body.
+	 * @param array             $headers      Cached headers.
+	 * @param int               $status_code  Cached status code.
+	 * @param DateTimeInterface $expiry       Expiry of the cached value.
+	 */
+	public function __construct( $body, array $headers, $status_code, DateTimeInterface $expiry ) {
```

Comment:
Don't use a type declaration with `array`, as that is broken in PHP. There's no possibility whatsoever to extend the `array` into an object with added logic and still pass that type declaration. Therefore, this is a hard lock-in into the exact `array` type, without any leeway.
```suggestion
	public function __construct( $body, $headers, $status_code, DateTimeInterface $expiry ) {
```

Reply:
Done in https://github.com/ampproject/amp-wp/pull/4509/commits/e87ab886e321bb181adb3668db9fdf01c77bcee3.